design-documents/graph-ql/coverage/search.md

```diff
@@ -0,0 +1,82 @@
+# Queries
+
+```graphql
+type Query {
+    #Filter supports multiple clauses which will be wrapped in logical AND operator
+    productSearch(
+        phrase: String!,
+        "Desired size of the search result page"
+        pageSize: Int = 20,
+        currentPage: Int = 1,
+        filter: [SearchClauseInput],
+        sort: [ProductSearchSortInput]
+    ): ProductSearchResponse!
+}
+
+input ProductSearchSortInput
+{
+    attribute: String!
+    direction: SortEnum!
+}
+
+# If from or to fields are omitted, $gte or $lte filter will be applied
+input SearchRangeInput {
+    from: Float
+    to: Float
+}
+
+input SearchClauseInput {
+    attribute_code: String!
+    in: [String]
+    eq: String
+    range: SearchRangeInput
+}
+
+type ProductSearchResponse {
+    items: [ProductSearchItem]
+    facets: [Aggregation]
+    facets_values: [Aggregation]
+    suggestions: [String]
+    related_terms: [String]
+    page_info: SearchResultPageInfo
+    total_count: Int
+}
+
+type ProductSearchItem {
+    product: ProductInterface!
+    highlights: [Highlight]
+}
+
+type Highlight {
+    attribute: String!
+    value: String!
+    matched_words: [String]!
+}
+
+interface Bucket {
+    #Human readable bucket title
+    title: String!
```

Comment:
All fields under the interface (e.g. Bucket) must be listed under any type that implements the interface. e.g. I don't see `title` listed under StatsBucket, ScalarBucket and RangeBucket.

Reply:
Add `attribute` under Bucket interface. This is so the client can use it on the subsequent requests (when user filters by facets). Without it, the client will not have an easy way to find the attribute value to use to begin with.